lib/tstracker/tracker.go

```diff
@@ -0,0 +1,276 @@
+package tstracker
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"time"
+
+	"github.com/jackc/pgx/v4"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-address"
+	pgchainbs "github.com/filecoin-project/go-bs-postgres-chainnotated"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/ipfs/go-cid"
+)
+
+func (tcs *tcs) CurrentDbTipSetKey(ctx context.Context) (*types.TipSetKey, abi.ChainEpoch, error) {
+
+	tskBytes, epoch, err := tcs.CurrentFilTipSetKeyBytes(ctx)
+	if err != nil {
+		return nil, -1, err
+	}
+
+	tsk, err := types.TipSetKeyFromBytes(tskBytes)
+	if err != nil {
+		return nil, -1, err
+	}
+
+	return &tsk, epoch, nil
+}
+
```

Comment:
Really the only thing you need from this file is this function, but doesn't hurt to c/p